File: config/route.ts
interface RouteModel {
    ts: () => Promise<any>,
    fun: string,
    permissions?: string[]
}

export const routes: Map<string, RouteModel> = new Map([
    ['login', { ts: () => import("@/app/api/auth"), fun: "login" }],
    ["logout", { ts: () => import("@/app/api/auth"), fun: "logout" }],
    ["checkToken", { ts: () => import("@/app/api/auth"), fun: "checkToken" }],
    ["updatePassword", { ts: () => import("@/app/api/auth"), fun: "updatePassword" }],
    ["getAdmin", { ts: () => import("@/app/api/auth"), fun: "getAdmin" }],
    ["updateAdminBySelf", { ts: () => import("@/app/api/auth"), fun: "updateAdminBySelf" }],
    ["getGoogleAuthQr", { ts: () => import("@/app/api/auth"), fun: "getGoogleAuthQr" }],
    ["verifyGoogleAuth", { ts: () => import("@/app/api/auth"), fun: "verifyGoogleAuth" }],
    ["resetGoogleAuthQr", { ts: () => import("@/app/api/auth"), fun: "resetGoogleAuthQr" }],
    ["cancelGoogleAuth", { ts: () => import("@/app/api/auth"), fun: "cancelGoogleAuth" }],

    ['getUsers', { ts: () => import("@/app/api/users"), fun: "get", permissions: ['admin', 'table'] }],
    ["insertUser", { ts: () => import("@/app/api/users"), fun: "insert", permissions: ['admin', 'table'] }],
    ["updateUser", { ts: () => import("@/app/api/users"), fun: "update", permissions: ['admin', 'table'] }],
    ["deleteUsers", { ts: () => import("@/app/api/users"), fun: "del", permissions: ['admin', 'table'] }],
    ["exportUsers", { ts: () => import("@/app/api/users"), fun: "exp", permissions: ['admin', 'table'] }],

    ['getRoles', { ts: () => import("@/app/api/roles"), fun: "get", permissions: ['admin', 'adminRoles'] }],
    ["updateRole", { ts: () => import("@/app/api/roles"), fun: "update", permissions: ['admin', 'adminRoles'] }],
    ["deleteRole", { ts: () => import("@/app/api/roles"), fun: "del", permissions: ['admin', 'adminRoles'] }],
    ["insertRole", { ts: () => import("@/app/api/roles"), fun: "insert", permissions: ['admin', 'adminRoles'] }],
    ["getRolePermission", { ts: () => import("@/app/api/roles"), fun: "getRolePermission", permissions: ['admin', 'adminRoles'] }],
    ["getAllRoles", { ts: () => import("@/app/api/roles"), fun: "getAll", permissions: ['admin', 'adminRoles', 'adminList'] }],

    ['getAdmins', { ts: () => import("@/app/api/admin"), fun: "get", permissions: ['admin', 'adminList'] }],
    ["updateAdmin", { ts: () => import("@/app/api/admin"), fun: "update", permissions: ['admin', 'adminList'] }],
    ["deleteAdmin", { ts: () => import("@/app/api/admin"), fun: "del", permissions: ['admin', 'adminList'] }],
    ["insertAdmin", { ts: () => import("@/app/api/admin"), fun: "insert", permissions: ['admin', 'adminList'] }],
    ["getAdminRole", { ts: () => import("@/app/api/admin"), fun: "getAdminRole", permissions: ['admin', 'adminList'] }],

    ["upload", { ts: () => import("@/app/api/common"), fun: "upload" }],
    ["getNotices", { ts: () => import("@/app/api/common"), fun: "getNotices" }],
    ["getNewNotices", { ts: () => import("@/app/api/common"), fun: "getNewNotices" }],
    ["getSysInfo", { ts: () => import("@/app/api/common"), fun: "getSysInfo" }],
    ["getAllConfig", { ts: () => import("@/app/api/common"), fun: "getAllConfig" }],
    ["updateConfig", { ts: () => import("@/app/api/common"), fun: "updateConfig" }],
    ["getDefaultLanguage", { ts: () => import("@/app/api/common"), fun: "getDefaultLanguage" }],

    ["easyQuery", { ts: () => import("@/app/api/easy-query"), fun: "query" }],
]);

File: backend/internal/router/route.go
package router

import (
	"nextjs-admin-go/internal/controller"
	"nextjs-admin-go/internal/middleware"

	"github.com/gin-gonic/gin"
)

func RegisterRoutes(r *gin.Engine, bc *controller.BaseController) {
	api := r.Group("/api")
	api.POST("/login", bc.Login)
	api.POST("/getDefaultLanguage", bc.GetDefaultLanguage)

	api.Use(middleware.AuthRequired(bc.DB))
	{
		api.POST("/checkToken", bc.CheckToken)
		api.POST("/logout", bc.Logout)
		api.POST("/updatePassword", bc.UpdatePassword)
		api.POST("/getAdmin", bc.GetAdmin)
		api.POST("/updateAdminBySelf", bc.UpdateAdminBySelf)
		api.POST("/getGoogleAuthQr", middleware.PermissionMiddleware([]string{"admin", ""}, bc.DB), bc.GetGoogleAuthQr)
		api.POST("/verifyGoogleAuth", bc.VerifyGoogleAuth)
		api.POST("/resetGoogleAuthQr", bc.ResetGoogleAuthQr)
		api.POST("/cancelGoogleAuth", bc.CancelGoogleAuth)

		api.POST("/upload", bc.Upload)
		api.POST("/getNotices", bc.GetNotices)
		api.POST("/getNewNotices", bc.GetNewNotices)
		api.POST("/getSysInfo", bc.GetSysInfo)
		api.POST("/getAllConfig", bc.GetAllConfig)
		api.POST("/updateConfig", bc.UpdateConfig)

		api.POST("/getUsers", bc.GetUsers)
		api.POST("/insertUser", bc.InsertUser)
		api.POST("/updateUser", bc.UpdateUser)
		api.POST("/deleteUsers", bc.DeleteUser)
		api.POST("/exportUsers", bc.ExportUsers)

		api.POST("/getRoles", bc.GetRoles)
		api.POST("/insertRole", bc.InsertRole)
		api.POST("/updateRole", bc.UpdateRole)
		api.POST("/deleteRole", bc.DeleteRole)
		api.POST("/getRolePermission", bc.GetRolePermission)
		api.POST("/getAllRoles", bc.GetAllRoles)

		api.POST("/getAdmins", bc.GetAdmins)
		api.POST("/insertAdmin", bc.InsertAdmin)
		api.POST("/updateAdmin", bc.UpdateAdmin)
		api.POST("/deleteAdmin", bc.DeleteAdmin)
		api.POST("/getAdminRole", bc.GetAdminRole)

		api.POST("/easyQuery", bc.EasyQuery)
	}
}
